### PR TITLE
Add node affinity

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- Adds node affinity for all deployments and cronjobs

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
- Jira cards
    - [PLAT-1252](https://airqoteam.atlassian.net/browse/PLAT-1252?atlOrigin=eyJpIjoiOGM3YjdmMzdhY2FmNDdkMmJmYzQ4OTI2ZDQyZGIwMWIiLCJwIjoiaiJ9)
    - [PLAT-1254](https://airqoteam.atlassian.net/browse/PLAT-1254?atlOrigin=eyJpIjoiOTAwNThiZTcyYjk3NDIwMzgwMGJlZTBlNGVlZDRjOGYiLCJwIjoiaiJ9)

**_HOW DO I TEST OUT THIS PR?_**
- Apply any of manifests to create a deployment/cronjob.
- Take note of the respective node affinity
- Confirm that the deployment/cronjob is running on the specified node by running the command below
    `kubectl get pods --output=wide`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- n/a

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- n/a

**_ARE THERE ANY RELATED PRs?_**
- [This frontend PR](https://github.com/airqo-platform/AirQo-frontend/pull/787)

### CODE REVIEW

**Comment by 123MwanjeMike** [comment]
Are all [these connectors](https://github.com/airqo-platform/AirQo-api/tree/ft-nodeAffinity/k8s/connectors) still in use or they need some clean up? cc @NoahNsimbe @Baalmart

File: k8s/data-mgt/stage-data-mgt-api.yaml [hunk 1]
```diff
@@ -19,6 +19,23 @@ spec:
       labels:
         app: airqo-stage-data-mgt-api
     spec:
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 10
```
Comment (123MwanjeMike) [inline k8s/data-mgt/stage-data-mgt-api.yaml:25]: Preference for deployment goes to the general-purpose nodes takes precedence (with a weight of 10), followed by high-memory nodes (with a weight of 1), and then any other nodes

File: k8s/device-monitor/stage-device-monitor-api.yaml [hunk 2]
```diff
@@ -41,11 +58,12 @@ spec:
                 name: env-device-monitoring-staging
           resources:
             requests:
-              cpu: 100m
-              memory: 250Mi
+              # Average usage was 0.0001 vCPU and 35Mi over the last 30 days
```
Comment (123MwanjeMike) [inline k8s/device-monitor/stage-device-monitor-api.yaml:61]: Data collected from past 30 days of observing

File: k8s/device-monitor/stage-device-monitor-api.yaml [hunk 3]
```diff
@@ -78,7 +96,7 @@ spec:
   metrics:
   - type: Resource
     resource:
-      name: cpu
+      name: memory
```
Comment (123MwanjeMike) [inline k8s/device-monitor/stage-device-monitor-api.yaml:99]: Scale based of memory usage

File: k8s/device-status/stage-airqo-device-status-job.yaml [hunk 4]
```diff
@@ -30,12 +40,9 @@ spec:
                 - configMapRef:
                     name: env-device-status-staging
               resources:
-                requests:
-                  cpu: 20m
-                  memory: 50Mi
                 limits:
-                  cpu: 2000m
-                  memory: 2Gi
+                  cpu: 1000m
```
Comment (123MwanjeMike) [inline k8s/device-status/stage-airqo-device-status-job.yaml:44]: 1vCPU

File: k8s/device-status/stage-airqo-device-status-job.yaml [hunk 5]
```diff
@@ -30,12 +40,9 @@ spec:
...
+                  memory: 4000Mi
```
Comment (123MwanjeMike) [inline k8s/device-status/stage-airqo-device-status-job.yaml:45]: 4GB RAM

File: .github/workflows/deploy-apis-to-staging.yml [hunk 6]
```diff
@@ -159,10 +159,6 @@ jobs:
               echo "::set-output name=run_meta_data::true"
             fi
 
-            if [[ $file == k8s/meta-data/* ]]; then
-              echo "::set-output name=run_meta_data::true"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-staging.yml:163]: Remove duplicate of the above

File: .github/workflows/deploy-apis-to-staging.yml [hunk 7]
```diff
@@ -577,7 +573,7 @@ jobs:
 
       - name: Deploy to K8S
         run: |
-          kubectl apply -f k8s/kafka-operator/namespace.yaml
+          kubectl apply -f k8s/namespaces/message-broker.yaml
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-staging.yml:576]: Use message-broker namespace definition in namespace folder.

File: k8s/test-deploy/stage-temp-device-registry-api.yaml [hunk 8]
```diff
@@ -1,51 +0,0 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: airqo-temp-stage-device-registry-api
-  namespace: staging
```
Comment (123MwanjeMike) [inline k8s/test-deploy/stage-temp-device-registry-api.yaml:5]: This was a test.

File: k8s/kafka-operator/namespace.yaml [hunk 9]
```diff
@@ -1,4 +0,0 @@
-apiVersion: v1
-kind: Namespace
```
Comment (123MwanjeMike) [inline k8s/kafka-operator/namespace.yaml:2]: This was a duplicate definition.

File: workflow-trigger [hunk 10]
```diff
@@ -1 +1 @@
-2022-08-28T20:00:00Z
\ No newline at end of file
+2022-09-02T20:00:00Z
```
Comment (NoahNsimbe) [inline workflow-trigger]: Hi @123MwanjeMike . Could we avoid this approach and setup a workflow for the changes in the k8s folders. Example
https://github.com/airqo-platform/AirQo-api/blob/b3c3fe4057fd438cc7afb12544dded1697006121/.github/workflows/deploy-apis-to-staging.yml#L186

Comment (123MwanjeMike) [reply workflow-trigger]: Thanks @NoahNsimbe 
Let me set that up asap

Comment (123MwanjeMike) [reply workflow-trigger]: This has been addressed?

**Review by Baalmart — APPROVED** [review]
LGTM

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Pull request merged**